app/magtape/test/test_slack.py

```diff
@@ -64,6 +85,179 @@ def test_slack_url_sub(self):
         value = magtape.slack_url_sub(self.slack_def_url)
         self.assertEqual(value, self.sub_value)
 
+    @patch("magtape.magtape.config.load_incluster_config")
+    @patch("magtape.magtape.config.load_kube_config")
+    @patch("magtape.magtape.client.CoreV1Api")
+    def test_get_namespace_slack_no_in_cluster_k8s_config(
+        self, mock_k8s_corev1api, mock_k8s_config, mock_k8s_config_in_cluster
+    ):
+
+        """Method to test reading secret containing Slack Incoming Webhook with no in-cluster k8s config"""
+
+        mock_k8s_config_in_cluster.side_effect = config.ConfigException
+        mock_k8s_api = mock_k8s_corev1api.return_value
+        mock_k8s_api.read_namespaced_secret.return_value = self.mock_slack_secret()
+        expected_result = self.test_slack_webhook_url
+
+        value = magtape.get_namespace_slack(
+            self.request_namespace, self.slack_webhook_secret,
+        )
+
+        self.assertRaises(config.ConfigException)
+        self.assertEqual(value, expected_result)
+        mock_k8s_config_in_cluster.assert_called_once_with()
+        mock_k8s_config.assert_called_once_with()
+        mock_k8s_api.read_namespaced_secret.assert_called_once_with(
+            self.slack_webhook_secret, self.request_namespace
+        )
+
+    @patch("magtape.magtape.config.load_incluster_config")
+    @patch("magtape.magtape.config.load_kube_config")
+    @patch("magtape.magtape.client.CoreV1Api")
+    def test_get_namespace_slack_no_k8s_config(
+        self, mock_k8s_corev1api, mock_k8s_config, mock_k8s_config_in_cluster
+    ):
+
+        """Method to test reading secret containing Slack Incoming Webhook with no k8s config"""
+
+        mock_k8s_config.side_effect = config.ConfigException
+        mock_k8s_config_in_cluster.side_effect = config.ConfigException
+        mock_k8s_api = mock_k8s_corev1api.return_value
+        mock_k8s_api.read_namespaced_secret.return_value = self.mock_slack_secret()
+
+        with self.assertRaises(Exception):
+
+            magtape.get_namespace_slack(
+                self.request_namespace, self.slack_webhook_secret,
+            )
+
+        self.assertRaises(config.ConfigException)
+        mock_k8s_config.assert_called_once_with()
+        mock_k8s_config_in_cluster.assert_called_once_with()
+
```

Comment:
`mock_k8s_corev1api.assert_not_called()` could probably be added to assert that logic isn't introduced which makes an unneeded call at some point in the case when the call should fail anyway because there's no k8s configs present.

Reply:
Good callout, will add this

Reply:
Done.